### DISCUSSION TITLE
What data values ​​and functions control the device?

### DISCUSSION THREAD

**Discussion by VengefulHat** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#798] [Q&A]
Hello, first I wanted to praise all the creators of this software and device. Good job.

I'm creating a post because I want to use the code to control BestWay-Lay-Z-SPA but using the Supla application. Many devices already work on this automation, I would also like to be able to control the jacuzzi using Supla, my device is in the model table (S100101).

I would like to ask if it is possible to suggest me how to easily, even through a regular Arduino UNO, connect and communicate (issue commands using UART)?
Is this fragment of code available somewhere, or do you not share it? I looked in the repository and could not find it.

If it is available somewhere, can you please point me to it? And could I ask for a simple code for Arduino UNO to control jacuzzi functionalities?

Why Arduino? I have built many projects on this platform (mainly UNO and ATMega328), I build my own systems for it and I understand it best, so if the code is difficult, it will definitely be easier for me to understand it on the Arduino module, which I know very well.

Thank you in advance for your interest and best regards.

**Comment by visualapproach** [comment]
Thanks! I have no knowledge of supla, so I can't help you with that. But you didn't ask me to. You wanted to run the core software on an arduino. Arduino is 10x slower than an ESP8266 with 40x less memory. Is it even possible? Well, as a matter of fact this project started on an Arduino UNO. I later added an ESP8266 module (ESP1?) to get basic wireless ability. The project has grown a bit since then LOL. I found some dusty ancient files in my drawer that I'll post and let you do some archeology on. I don't know if it works as is or anything, but it should be relatively straightforward to go through the code.
Here is a picture of my first "module".
![IMG_3762](https://github.com/user-attachments/assets/fdddb840-2c6d-4c9b-8ee9-fff5c4cece04)

And the code. There is one program for the ESP and one for the UNO. You wanna look at the UNO file.
[arduino bwc.zip](https://github.com/user-attachments/files/16968787/arduino.bwc.zip)
